rmcreyes/cpen391-user-app
# Connect Car api

# :ship: Pull Request

## :question: Purpose

Connected user app with Car Apis for the Car Pages
- Added functionality to add, edit, delete cars

## :hammer: Validation Strategy

Manual testing steps
- log in to the app, and navigate to the Cars page using the bottom navigation
- Verify that if there are no cars registered under the account the page shows the "empty" message. Else verify that all the registered cars are displayed. 
- Click the "Add" button to navigate to the "add car" page
- Enter a valid (6 character) license plate, and a car name, and click the "Add" button
    - verify that if an invalid license plate entered, then the app notifies the error
    - verify that if the process was successful, you will navigate back to the cars page, with the new item showing up in the horizontal list. 
    - verify that if no car name was entered, the name will be saved to be the same as the license plate
 - Click on any car item,  to navigate to the "edit" page for the car item. 
 - Verify that you are able to "edit" the car name, but not the license plate. 
 - Click on the "delete" button, and verify that you navigate back to the cars page, and the deleted item is no longer part of the cars list. 

## :tickets: Ticket(s)

Closes #9
